dashboard/app/helpers/levels_helper.rb

```diff
@@ -348,7 +348,7 @@ def set_tts_options(level_options, app_options)
   end
 
   def set_hint_prompt_options(level_options)
-    if @script && @script.hint_prompt_enabled?
+    if @script && @script.csf? && @level.hint_prompt_attempts_threshold
```

Comment:
I think I'd like to keep `hint_prompt_enabled?` around for clarity; we can update it to just alias `csf?` for now, but I suspect we'll be wanting to adjust that again in the future